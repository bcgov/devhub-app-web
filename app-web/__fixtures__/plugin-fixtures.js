/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
export const SOURCE_DEVHUB_GITHUB_QL_NODE = {
  id: 'd90c1b12-9055-580b-adb3-b29354871657',
  source: 'design-system',
  sourceName: 'Design System',
  sourcePath: 'https://github.com/bcgov/design-system/',
  resourcePath: '/design-system/README_123o8123',
  path: 'components/footer/README.md',
  owner: 'bcgov',
  labels: ['Design System', 'Components'],
  fileName: 'about.md',
  fileType: 'Markdown',
  pagePath: '/design-system/about_G0Z6oO3_3',
  originalSource:
    'https://github.com/bcgov/design-system/blob/master/components/footer/something/README.md',
  internal: {
    mediaType: 'text/markdown',
    type: 'SourceDevhubGithub',
    owner: 'gatsby-source-github-all',
    contentDigest: '1239087asdfjkas',
    content: '---\nsomething:true\n---\n# Hello world\n',
  },
  childMarkdownRemark: {
    id: 'c03ba7af-a099-5dbe-8aec-5922922e0fcc >>> MarkdownRemark',
    html: '<h1>Hello World</h1>',
    frontmatter: {
      something: 'true',
      _PARENT: 'c03ba7af-a099-5dbe-8aec-5922922e0fcc',
      parent: 'c03ba7af-a099-5dbe-8aec-5922922e0fcc',
      status: null,
    },
    headings: [
      {
        value: 'Hello World',
        depth: 1,
      },
    ],
    internal: {
      content:
        '---\ndescription: What the Design System is and how it works.\ntitle: About\n---\n\n## About\n\nThe Design System helps developers and designers build better digital products and services.\n\nIt’s a collection of digital resources and tools – including a library of reusable UI/interface components and service patterns. The system makes it easier and faster to build custom B.C. government websites and applications. \n\nComponents are collectively built by the government community, meet accessibility standards and are open for input and improvement.\n\nUse the Design System if you’re building custom B.C. government and applications. For more information about applying for external URLs, contact the B.C. government Joint Working Group.\n\n## The Team\n\nThe Design System is managed by a team of designers and developers and the government community. The system is a part of the DevHub, a project by the Developers’ Exchange that brings together UI/interface and back-end resources for digital teams.\n\nKelsey Singbeil – Team Lead, Service Design\n\nBryan Smith – Service Design Specialist\n\nOri Nevares – Service Design Specialist\n\n## The Community\n\nThe system is collectively built and maintained by the government community. It’s open source, which means that anybody can provide input, offer suggestions and get involved.\n\nShare your feedback to make this resource better for everyone.\n\n## How the system works\n\nHow might we improve people’s interaction with B.C. government digital products and services?\n\nThe system is a part of the answer.\n\nIt provides developers and designers with the tools they need to build great digital government products and services. The system is collectively built and managed, so all the tools and resources are the product of the community.\n\nAs part of this, we are committed to making all the tools and resources we create:\n\n**Maintained**  \nThe community and Design System management team will support the maintenance and development of new resources and tools. This means you can flag issues and they’ll be addressed, discuss rationale and directly chat with the management team whenever you need.\n\n**Collaborative**\nThe system is only valuable if the community finds it valuable. If a tool or resource isn’t meeting your needs, tell us why and work with us to make it better. We’ll also come to you – in your office, at your desk and share what’s new, ask what you need and share how you can get involved.\n\n**Open**\nThe system is built in GitHub and meant to be shared and improved by everyone. The code is open source and our work is in the open. We are transparent about how choices are made and tools are improved. Ask us…anything.\n\n**Contribute to the Design System**\n\n***Help with design***\n-   Suggest a component\n-   Improve documentation\n\n***Help with code***\n-  Code contribution guide\n-   Report an issue\n\n***Help with usability***\n-   Component discussions\n-   Provide research findings\n\n',
      type: 'MarkdownRemark',
      contentDigest: 'b186249c65e12e08344e31e289d99ab8',
      owner: 'gatsby-transformer-remark',
    },
  },
};

export const FILE_QL_NODE = {
  id: '/foo/Index.test.js absPath of file',
  internal: {
    contentDigest: '2a9e4d519252bf9f9ccc9a931d2d731e',
    mediaType: 'application/javascript',
    type: 'File',
    description: 'File "src/pages/Index.test.js"',
    owner: 'gatsby-source-filesystem',
  },
  birthtime: '2018-10-24T21:16:43.521Z',
  publicURL: '/static/Index.test-2a9e4d519252bf9f9ccc9a931d2d731e.js',
  ctime: '2018-10-24T21:16:43.522Z',
  mtime: '2018-10-24T21:16:43.522Z',
  atime: '2018-10-25T20:32:50.743Z',
  birthtimeMs: 1540415803521.444,
  ctimeMs: 1540415803521.575,
  mtimeMs: 1540415803521.575,
  atimeMs: 1540499570743.2075,
  sourceInstanceName: 'pages',
  absolutePath: '/foo/Index.test.js',
  relativePath: 'Index.test.js',
  relativeDirectory: '',
  extension: 'js',
  size: 1674,
  prettySize: '1.67 kB',
  name: 'Index.test',
  ext: '.js',
  base: 'Index.test.js',
  dev: 16777220,
  mode: 33188,
  nlink: 1,
  uid: 323756479,
  gid: 1839645156,
  blocks: 8,
  ino: 11739371,
  blksize: 4194304,
  rdev: 0,
};
