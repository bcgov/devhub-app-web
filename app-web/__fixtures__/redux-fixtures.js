/*
Copyright 2018 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
import * as actions from '../src/store/actions/actionTypes';
export const ACTIONS = {
  LOAD_SIPHON_NODES: {
    type: actions.LOAD_SIPHON_NODES,
    payload: {
      nodes: [{ foo: 'bar' }, { foo: 'baz' }, { foo: 'foo' }],
    },
  },
  FILTER_SIPHON_NODES: {
    type: actions.FILTER_SIPHON_NODES,
    payload: {
      filteredBy: 'foo',
      value: 'bar',
    },
  },
  FILTER_SIPHON_NODES_BY_ALL: {
    type: actions.FILTER_SIPHON_NODES,
    payload: {
      filteredBy: 'foo',
      value: 'All',
    },
  },
  SELECT_FILTER_OPTION: {
    type: actions.SET_SELECTED_FILTER_OPTION,
    payload: {
      foo: 'bar',
    },
  },
  TOGGLE_MAIN_NAVIGATION_ON: {
    type: actions.TOGGLE_MAIN_NAVIGATION,
    payload: {
      toggled: true,
    },
  },
  TOGGLE_MAIN_NAVIGATION_OFF: {
    type: actions.TOGGLE_MAIN_NAVIGATION,
    payload: {
      toggled: false,
    },
  },
};

export const INITIAL_STATES = {
  SIPHON: {
    nodes: [],
    filteredNodes: [],
    groupBy: null,
    loading: false,
    error: false,
    messages: [],
  },
  UI: {
    selectedFilterOption: null,
    mainNavigationToggled: false,
  },
};
