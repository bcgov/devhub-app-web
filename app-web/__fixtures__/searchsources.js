/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/
// run time apollo query fixtures

export const ROCKET_CHAT = [
  {
    id: '1', 
    roomId: '2', author: 'apple', time: '2019-02-12', message: 'apples are clearly superior',
  },
  {
    id: '2', 
    roomId: '2', author: 'orange', time: '2019-02-12', message: 'Nice joke apple, are trying to perform some time of standup?',
  },
  {
    id: '3', 
    roomId: '2', author: 'apple', time: '2019-02-12', message: 'Don\'t play coy with me orange, you know I\'m telling the truth',
  },
  {
    id: '4', 
    roomId: '2', author: 'orange', time: '2019-02-12', message: 'I ain\'t never hurd of people curing scurvy with apples yo. Hook line and sinker dawg.',
  },
]