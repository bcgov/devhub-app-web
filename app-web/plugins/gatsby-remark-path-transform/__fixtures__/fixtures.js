const PARAGRAPH_AST = {
    'type': 'paragraph',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
  };

const IMAGE_AST_ABSOLUTE = {
    'type': 'image',
    'url': 'https://www.google.com/images/dog.png',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
};

const IMAGE_AST_RELATIVE = {
    'type': 'image',
    'url': '../images/dog.png',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
};

const LINK_AST_ABSOLUTE = {
    'type': 'link',
    'url': 'https://www.google.com/images/dog.png',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
};

const LINK_AST_RELATIVE = {
    'type': 'link',
    'url': '../images/dog.png',
    'children': [
      {'type': 'text', 'value': 'GITHUB-EMBED GITHUB-EMBED'}
    ],
};

const GRAPH_QL_PARENT_NODE = {
    internal: {
        type: 'node',
        mediaType: 'text',
    },
    parent: null,
};
const MARKDOWN_NODE = {
    internal: {
        type: 'node',
        mediaType: 'text',
    },
    parent: null,
};
module.exports = {
    PARAGRAPH_AST,
    IMAGE_AST_ABSOLUTE,
    IMAGE_AST_RELATIVE,
    LINK_AST_ABSOLUTE,
    LINK_AST_RELATIVE,
    GRAPH_QL_PARENT_NODE,
    MARKDOWN_NODE,
};
