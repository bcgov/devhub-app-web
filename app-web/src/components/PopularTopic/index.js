import React from 'react';
import PropTypes from 'prop-types';
import { POPULAR_TOPIC_CONFIGURATION, DYNAMIC_TOPIC_PATHS } from '../../constants/ui';
import Topic from '../Topic';

const PopularTopic = ({ nodes }) =>
  nodes.length > 0 ? (
    <Topic
      resources={nodes.map(n => ({
        ...n,
        path: `/topic/${DYNAMIC_TOPIC_PATHS.popular}/${n.fields.slug}`,
      }))}
      title={POPULAR_TOPIC_CONFIGURATION.name}
      description={POPULAR_TOPIC_CONFIGURATION.description}
      link={{ to: `/topic/${DYNAMIC_TOPIC_PATHS.popular}`, text: 'Popular' }}
    />
  ) : null;

PopularTopic.propTypes = {
  nodes: PropTypes.arrayOf(
    PropTypes.shape({
      path: PropTypes.string,
      fields: PropTypes.shape({
        title: PropTypes.string,
        description: PropTypes.string,
        image: PropTypes.string,
        slug: PropTypes.string,
        labels: PropTypes.array,
        tags: PropTypes.array,
        pagePaths: PropTypes.array,
      }),
    }),
  ),
};
export default PopularTopic;
