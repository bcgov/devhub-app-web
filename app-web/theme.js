/*
Copyright 2019 Province of British Columbia

Licensed under the Apache License, Version 2.0 (the "License");
you may not use this file except in compliance with the License.
You may obtain a copy of the License at 

   http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing, software
distributed under the License is distributed on an "AS IS" BASIS,
WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
See the License for the specific language governing permissions and
limitations under the License.

Created by Patrick Simonian
*/

/*
Emotion theming object

We are transitioning from common elements such as colors, background colors being abstracted away into
a themes provider. This is a very common practice now a days and makes it maintain themes on the fly
docs on theming is here https://emotion.sh/docs/theming#___gatsby

Docs on emotion https://emotion.sh/docs/introduction

--- our idea is to introduce emotion early and slowly replace css modules with emotion styled components
when and as needed. This is a slow processes but will lead to better and more maintaible styling in the long run.
*/

import { SPACING } from './src/constants/designTokens';
import { RESOURCE_TYPES, TOPICS, SEARCH_RESOURCE_TYPES, JOURNEY } from './src/constants/ui';

export const DEVHUB_PALETTE = {
  [RESOURCE_TYPES.COMPONENTS]: '#4299D2',
  [RESOURCE_TYPES.DOCUMENTATION]: '#246BD1',
  [RESOURCE_TYPES.PEOPLE]: '#E17039',
  [RESOURCE_TYPES.SELF_SERVICE_TOOLS]: '#F3BA45',
  [TOPICS]: '#444',
  [RESOURCE_TYPES.REPOSITORIES]: '#6e5494',
  [SEARCH_RESOURCE_TYPES.GITHUB_ISSUE]: '#6e5494',
  [RESOURCE_TYPES.EVENTS]: '#f44b42',
  [JOURNEY]: '#009688',
  lightgreen: '#66bb6a',
  yellow: '#FCBA19',
  lightyellow: '#F3BA45',
  grey: '#494949',
  lightgrey: '#F2F2F2',
  darkgrey: '#444',
  orange: '#E17039',
  purple: '#6E5494',
  red: '#D8292F',
  lightred: '#F44B42',
  green: '#2E8540',
  blue: '#003366',
  babyblue: '#4299D2',
  lightblue: '#246BD1',
  linkblue: '#1A5A96',
  bgBlue: '#38598A',
  white: '#ffffff',
};

export default {
  colors: {
    ...DEVHUB_PALETTE,
  },
  padding: {
    main: {
      mobile: `${SPACING['2x']} ${SPACING['2x']}`,
      desktop: `${SPACING['2x']} ${SPACING['13x']}`,
    },
  },
  breakpoints: {
    // specific breakpoints
    main: {
      desktop: '@media screen and (min-width: 932px)',
    },
  },
};
