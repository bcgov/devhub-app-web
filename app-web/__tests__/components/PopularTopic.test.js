import React from 'react';
import { render } from 'react-testing-library';
import { ThemeProvider } from 'emotion-theming';
import theme from '../../theme';
import PopularTopic from '../../src/components/PopularTopic';
import { GITHUB_RAW_1, GITHUB_RAW_2 } from '../../__fixtures__/nodes';

describe('PopularTopic', () => {
  it('matches snapshot', () => {
    const nodes = [GITHUB_RAW_1, GITHUB_RAW_2];

    const { container } = render(
      <ThemeProvider theme={theme}>
        <PopularTopic nodes={nodes} />
      </ThemeProvider>,
    );

    expect(container).toMatchSnapshot();
  });
});
