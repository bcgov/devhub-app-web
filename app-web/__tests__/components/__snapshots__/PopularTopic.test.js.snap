// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PopularTopic matches snapshot 1`] = `
.emotion-61 {
  max-width: 1100px;
  margin: 20px 0;
  border: 1px solid #ccc;
  border-top: transparent;
}

.emotion-0 {
  background-color: #444;
  -webkit-flex: 0 0 10px;
  -ms-flex: 0 0 10px;
  flex: 0 0 10px;
  min-height: 10px;
  min-height: 15px;
}

.emotion-59 {
  padding: 0 15px;
  padding-top: 10px;
  padding-bottom: 20px;
}

.emotion-4 {
  color: #444;
  font-size: 15px;
  margin-bottom: 0;
  margin-top: 0;
  font-weight: 400;
}

.emotion-2 {
  color: #444;
}

.emotion-9 {
  font-size: 18px;
  color: #494949;
  line-height: 1.3;
  margin-bottom: 0;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  text-transform: capitalize;
  word-break: break-word;
  font-weight: 700;
  margin: 10px 0;
  font-size: 26px;
}

.emotion-7 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #1A5A96;
  font-weight: 400;
  text-transform: capitalize;
  padding: 0 4px;
  margin: 0 2px;
  color: inherit;
  padding: 0px;
  margin: 0px;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.emotion-7:hover {
  color: inherit;
  -webkit-text-decoration-color: #1a5a96;
  text-decoration-color: #1a5a96;
}

.emotion-11 {
  line-height: 1.4;
  font-size: 14px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-bottom: 0;
  margin-top: 4px;
  text-transform: none;
  max-width: 500px;
  margin-bottom: 15px;
}

.emotion-32 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #1A5A96;
  font-weight: 400;
  text-transform: capitalize;
  padding: 0 4px;
  margin: 0 2px;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: initial;
}

.emotion-32:hover {
  -webkit-text-decoration: none;
  text-decoration: none;
  color: initial;
  cursor: pointer;
}

.emotion-29 {
  width: 265px;
  height: 238.5px;
  margin: 10px 0;
  border: 1px solid #ccc;
  border-top: transparent;
  border-radius: 2px;
  box-sizing: border-box;
  -webkit-transition: -webkit-transform 0.25s ease-out;
  -webkit-transition: transform 0.25s ease-out;
  transition: transform 0.25s ease-out;
  box-shadow: 0 1px 2px 1px #00000026;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
  overflow: hidden;
}

.emotion-29:hover {
  box-shadow: 0 2px 2px 1px #00000026;
  -webkit-transform: translateY(-2px);
  -ms-transform: translateY(-2px);
  transform: translateY(-2px);
}

.emotion-13 {
  background-color: #246BD1;
  -webkit-flex: 0 0 10px;
  -ms-flex: 0 0 10px;
  flex: 0 0 10px;
  min-height: 10px;
}

.emotion-27 {
  padding: 6px 10px;
  height: 100%;
  overflow: auto;
  height: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-flex-flow: column nowrap;
  -ms-flex-flow: column nowrap;
  flex-flow: column nowrap;
}

.emotion-15 {
  color: #246BD1;
}

.emotion-19 {
  font-size: 18px;
  color: #494949;
  line-height: 1.3;
  margin-bottom: 0;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  text-transform: capitalize;
  word-break: break-word;
}

.emotion-21 {
  line-height: 1.4;
  font-size: 14px;
  -webkit-flex: 0 0 auto;
  -ms-flex: 0 0 auto;
  flex: 0 0 auto;
  margin-bottom: 0;
  margin-top: 4px;
  text-transform: none;
}

.emotion-25 {
  -webkit-flex: 1 1 auto;
  -ms-flex: 1 1 auto;
  flex: 1 1 auto;
  -webkit-align-self: center;
  -ms-flex-item-align: center;
  align-self: center;
  overflow: hidden;
  padding-top: 10px;
  text-align: center;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
}

.emotion-23 {
  width: 100%;
  height: auto;
  object-fit: scale-down;
  max-height: 100%;
  margin-bottom: 0;
}

.emotion-34 {
  background-color: #4299D2;
  -webkit-flex: 0 0 10px;
  -ms-flex: 0 0 10px;
  flex: 0 0 10px;
  min-height: 10px;
}

.emotion-36 {
  color: #4299D2;
}

.emotion-57 {
  text-align: right;
  font-size: 1.15em;
}

.emotion-55 {
  -webkit-text-decoration: underline;
  text-decoration: underline;
  color: #1A5A96;
  font-weight: 400;
  text-transform: capitalize;
  padding: 0 4px;
  margin: 0 2px;
}

<div>
  <div
    class="emotion-61 emotion-62"
  >
    <div
      class="emotion-0 emotion-1"
      color="Topics"
    />
    <div
      class="emotion-59 emotion-60"
    >
      <div
        style="margin-bottom: 6px;"
      >
        <h3
          class="emotion-4 emotion-5"
        >
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-layer-group fa-w-16  emotion-2 emotion-3"
            data-icon="layer-group"
            data-prefix="fas"
            data-testid="resource-type-icon"
            focusable="false"
            role="img"
            type="Topics"
            viewBox="0 0 512 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M12.41 148.02l232.94 105.67c6.8 3.09 14.49 3.09 21.29 0l232.94-105.67c16.55-7.51 16.55-32.52 0-40.03L266.65 2.31a25.607 25.607 0 0 0-21.29 0L12.41 107.98c-16.55 7.51-16.55 32.53 0 40.04zm487.18 88.28l-58.09-26.33-161.64 73.27c-7.56 3.43-15.59 5.17-23.86 5.17s-16.29-1.74-23.86-5.17L70.51 209.97l-58.1 26.33c-16.55 7.5-16.55 32.5 0 40l232.94 105.59c6.8 3.08 14.49 3.08 21.29 0L499.59 276.3c16.55-7.5 16.55-32.5 0-40zm0 127.8l-57.87-26.23-161.86 73.37c-7.56 3.43-15.59 5.17-23.86 5.17s-16.29-1.74-23.86-5.17L70.29 337.87 12.41 364.1c-16.55 7.5-16.55 32.5 0 40l232.94 105.59c6.8 3.08 14.49 3.08 21.29 0L499.59 404.1c16.55-7.5 16.55-32.5 0-40z"
              fill="currentColor"
            />
          </svg>
          <span
            style="margin: 4px;"
          >
            Topic
          </span>
        </h3>
      </div>
      <div
        class="emotion-9 emotion-10"
      >
        <a
          class="emotion-6 emotion-7 emotion-8"
          data-testid="topic-Popular"
          href="/topic/popular"
        >
          Popular
        </a>
      </div>
      <div
        class="emotion-11 emotion-12"
      >
        These resources have been getting some attention lately
      </div>
      <div
        class="BrainhubCarousel"
      >
        <div
          class="BrainhubCarousel__trackContainer"
        >
          <ul
            class="BrainhubCarousel__track BrainhubCarousel__track--draggable"
            style="margin-left: 0px; width: 0px; transform: translateX(0px);"
          >
            <li
              class="BrainhubCarouselItem BrainhubCarouselItem--active"
              style="padding-right: 0px; padding-left: 0px; width: 275px; max-width: 275px; min-width: 275px;"
            >
              <a
                class="emotion-31 emotion-32 emotion-8"
                data-testid="ui-gatsby-link"
                href="/topic/popular/Introduction-To-Github-and-Gov"
              >
                <article
                  class="emotion-29 emotion-30"
                >
                  <div
                    class="emotion-13 emotion-14"
                    color="Documentation"
                  />
                  <div
                    class="emotion-27 emotion-28"
                  >
                    <div
                      style="margin-bottom: 6px;"
                    >
                      <h3
                        class="emotion-4 emotion-5"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-file fa-w-12  emotion-15 emotion-3"
                          data-icon="file"
                          data-prefix="fas"
                          data-testid="resource-type-icon"
                          focusable="false"
                          role="img"
                          type="Documentation"
                          viewBox="0 0 384 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M224 136V0H24C10.7 0 0 10.7 0 24v464c0 13.3 10.7 24 24 24h336c13.3 0 24-10.7 24-24V160H248c-13.2 0-24-10.8-24-24zm160-14.1v6.1H256V0h6.1c6.4 0 12.5 2.5 17 7l97.9 98c4.5 4.5 7 10.6 7 16.9z"
                            fill="currentColor"
                          />
                        </svg>
                        <span
                          style="margin: 4px;"
                        >
                          Documentation
                        </span>
                        <small>
                          <svg
                            aria-hidden="true"
                            class="svg-inline--fa fa-external-link-alt fa-w-18 "
                            data-icon="external-link-alt"
                            data-prefix="fas"
                            focusable="false"
                            role="img"
                            viewBox="0 0 576 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
                              fill="currentColor"
                            />
                          </svg>
                        </small>
                      </h3>
                    </div>
                    <div
                      class="emotion-19 emotion-20"
                    >
                      Introduction To Github and Gov
                    </div>
                    <p
                      class="emotion-21 emotion-22"
                    >
                      Everything you need to know about working with Github in the BC Government.
                    </p>
                    <div
                      class="emotion-25 emotion-26"
                    >
                      <img
                        alt="Introduction To Github and Gov"
                        class="emotion-23 emotion-24"
                        src="../images/octokat.png"
                      />
                    </div>
                  </div>
                </article>
              </a>
            </li>
            <li
              class="BrainhubCarouselItem"
              style="padding-right: 0px; padding-left: 0px; width: 275px; max-width: 275px; min-width: 275px;"
            >
              <a
                class="emotion-31 emotion-32 emotion-8"
                data-testid="ui-gatsby-link"
                href="/topic/popular/Introduction-To-Github-and-Gov"
              >
                <article
                  class="emotion-29 emotion-30"
                >
                  <div
                    class="emotion-34 emotion-14"
                    color="Components"
                  />
                  <div
                    class="emotion-27 emotion-28"
                  >
                    <div
                      style="margin-bottom: 6px;"
                    >
                      <h3
                        class="emotion-4 emotion-5"
                      >
                        <svg
                          aria-hidden="true"
                          class="svg-inline--fa fa-puzzle-piece fa-w-18  emotion-36 emotion-3"
                          data-icon="puzzle-piece"
                          data-prefix="fas"
                          data-testid="resource-type-icon"
                          focusable="false"
                          role="img"
                          type="Components"
                          viewBox="0 0 576 512"
                          xmlns="http://www.w3.org/2000/svg"
                        >
                          <path
                            d="M519.442 288.651c-41.519 0-59.5 31.593-82.058 31.593C377.409 320.244 432 144 432 144s-196.288 80-196.288-3.297c0-35.827 36.288-46.25 36.288-85.985C272 19.216 243.885 0 210.539 0c-34.654 0-66.366 18.891-66.366 56.346 0 41.364 31.711 59.277 31.711 81.75C175.885 207.719 0 166.758 0 166.758v333.237s178.635 41.047 178.635-28.662c0-22.473-40-40.107-40-81.471 0-37.456 29.25-56.346 63.577-56.346 33.673 0 61.788 19.216 61.788 54.717 0 39.735-36.288 50.158-36.288 85.985 0 60.803 129.675 25.73 181.23 25.73 0 0-34.725-120.101 25.827-120.101 35.962 0 46.423 36.152 86.308 36.152C556.712 416 576 387.99 576 354.443c0-34.199-18.962-65.792-56.558-65.792z"
                            fill="currentColor"
                          />
                        </svg>
                        <span
                          style="margin: 4px;"
                        >
                          Component
                        </span>
                        <small>
                          <svg
                            aria-hidden="true"
                            class="svg-inline--fa fa-external-link-alt fa-w-18 "
                            data-icon="external-link-alt"
                            data-prefix="fas"
                            focusable="false"
                            role="img"
                            viewBox="0 0 576 512"
                            xmlns="http://www.w3.org/2000/svg"
                          >
                            <path
                              d="M576 24v127.984c0 21.461-25.96 31.98-40.971 16.971l-35.707-35.709-243.523 243.523c-9.373 9.373-24.568 9.373-33.941 0l-22.627-22.627c-9.373-9.373-9.373-24.569 0-33.941L442.756 76.676l-35.703-35.705C391.982 25.9 402.656 0 424.024 0H552c13.255 0 24 10.745 24 24zM407.029 270.794l-16 16A23.999 23.999 0 0 0 384 303.765V448H64V128h264a24.003 24.003 0 0 0 16.97-7.029l16-16C376.089 89.851 365.381 64 344 64H48C21.49 64 0 85.49 0 112v352c0 26.51 21.49 48 48 48h352c26.51 0 48-21.49 48-48V287.764c0-21.382-25.852-32.09-40.971-16.97z"
                              fill="currentColor"
                            />
                          </svg>
                        </small>
                      </h3>
                    </div>
                    <div
                      class="emotion-19 emotion-20"
                    >
                      Introduction To Github and Gov
                    </div>
                    <p
                      class="emotion-21 emotion-22"
                    >
                      Everything you need to know about working with Github in the BC Government.
                    </p>
                    <div
                      class="emotion-25 emotion-26"
                    >
                      <img
                        alt="Introduction To Github and Gov"
                        class="emotion-23 emotion-24"
                        src="../images/octokat.png"
                      />
                    </div>
                  </div>
                </article>
              </a>
            </li>
          </ul>
        </div>
      </div>
      <div
        class="emotion-57 emotion-58"
      >
        <a
          class="emotion-55 emotion-8"
          data-testid="ui-gatsby-link"
          href="/topic/popular"
        >
          Popular
           
          <svg
            aria-hidden="true"
            class="svg-inline--fa fa-chevron-right fa-w-10 "
            data-icon="chevron-right"
            data-prefix="fas"
            focusable="false"
            role="img"
            viewBox="0 0 320 512"
            xmlns="http://www.w3.org/2000/svg"
          >
            <path
              d="M285.476 272.971L91.132 467.314c-9.373 9.373-24.569 9.373-33.941 0l-22.667-22.667c-9.357-9.357-9.375-24.522-.04-33.901L188.505 256 34.484 101.255c-9.335-9.379-9.317-24.544.04-33.901l22.667-22.667c9.373-9.373 24.569-9.373 33.941 0L285.475 239.03c9.373 9.372 9.373 24.568.001 33.941z"
              fill="currentColor"
            />
          </svg>
        </a>
      </div>
    </div>
  </div>
</div>
`;
